iot-hub/Samples/service/ImportExportDevicesSample/ImportExportDevicesSample.cs

```diff
@@ -308,37 +303,37 @@ private async Task GenerateAndAddDevices(string hubConnectionString, string cont
             await generatedListBlob.DeleteIfExistsAsync().ConfigureAwait(false);
 
             // Write list of serialized objects to the blob.
-            using (CloudBlobStream stream = await generatedListBlob.OpenWriteAsync().ConfigureAwait(false))
+            using CloudBlobStream stream = await generatedListBlob.OpenWriteAsync().ConfigureAwait(false);
+            byte[] bytes = Encoding.UTF8.GetBytes(sb.ToString());
+            for (var i = 0; i < bytes.Length; i += 500)
             {
-                byte[] bytes = Encoding.UTF8.GetBytes(sb.ToString());
-                for (var i = 0; i < bytes.Length; i += 500)
-                {
-                    int length = Math.Min(bytes.Length - i, 500);
-                    await stream.WriteAsync(bytes, i, length).ConfigureAwait(false);
-                }
+                int length = Math.Min(bytes.Length - i, 500);
+                await stream.WriteAsync(bytes, i, length).ConfigureAwait(false);
             }
+            stream.Commit();
```

Comment:
This was the key that made the sample start working. I found other places, like here, to add it as well.